tensorflow/core/kernels/lrn_op.cc

```diff
@@ -223,6 +226,142 @@ struct LaunchLRN<GPUDevice, T> {
 
 #endif  // GOOGLE_CUDA
 
+#ifdef TENSORFLOW_USE_SYCL
+
+namespace {
+
+template <typename T>
+struct LRNInvSYCL {
+  LRNInvSYCL(T) {}
+
+  inline T operator()(T in1, T in2) const {
+    return in1 / in2;
+  }
+};
+
+template <typename T>
+struct LRNRsqrtSYCL {
+  LRNRsqrtSYCL(T) {}
+
+  inline T operator()(T in1, T in2) const {
+    return in1 * cl::sycl::rsqrt(in2);
+  }
+};
+
+template <typename T>
+struct LRNGenSYCL {
+  LRNGenSYCL(T beta) : beta_(beta) {}
+
+  inline T operator()(T in1, T in2) const {
+    return in1 * cl::sycl::exp(cl::sycl::log(in2) * -beta_);
+  }
+
+ private:
+  T beta_;
+};
+
+template <typename T, typename BinaryOp>
+struct LRNKernelSYCL {
+  using r_acc =
+    cl::sycl::accessor<Eigen::buffer_scalar_t, 1,
+                       cl::sycl::access::mode::read,
+                       cl::sycl::access::target::global_buffer>;
+  using dw_acc =
+    cl::sycl::accessor<Eigen::buffer_scalar_t, 1,
+                       cl::sycl::access::mode::discard_write,
+                       cl::sycl::access::target::global_buffer>;
+
+  LRNKernelSYCL(int depth, int depth_radius, T bias, T alpha, T beta,
+                r_acc in_acc, dw_acc out_acc)
+    : depth_(depth), depth_radius_(depth_radius), bias_(bias), alpha_(alpha),
+      binary_op_(BinaryOp(beta)), in_acc_(in_acc), out_acc_(out_acc) {}
+
+  inline void operator()(cl::sycl::item<2> item) {
+    T* in_data = ConvertToActualTypeSycl(T, in_acc_);
+    T* out_data = ConvertToActualTypeSycl(T, out_acc_);
+
+    const auto id = item.get_linear_id();
+    const int row = item.get_id(0);
+    const int col = item.get_id(1);
+    const int base_id = row * depth_;
+    int start_col = col - depth_radius_;
+    int band_width = 2 * depth_radius_ + 1;
+    if (col < depth_radius_) {
+      start_col = 0;
+      band_width -= depth_radius_ - col;
+    } else if (start_col + band_width - 1 >= depth_) {
+      band_width = depth_ - start_col;
+    }
+    T sum = T(0);
+    T act_val;
```

Comment:
Declare and assign act_val inside the loop.
There is no difference in performance